web/src/lib/components/album-page/__tests__/album-card.spec.ts

```diff
@@ -1,103 +1,99 @@
-      count: 5,
-      shared: false,
-    },
-    {
-      album: albumFactory.build({ albumThumbnailAssetId: null, shared: true, assetCount: 2 }),
-      count: 2,
-      shared: true,
-    },
-  ])('shows album data without thumbnail with count $count - shared: $shared', async ({ album, count, shared }) => {
-    sut = render(AlbumCard, { album });
-
-    const albumImgElement = sut.getByTestId('album-image');
-    const albumNameElement = sut.getByTestId('album-name');
-    const albumDetailsElement = sut.getByTestId('album-details');
-    const detailsText = `${count} items` + (shared ? ' . Shared' : '');
-
-    expect(albumImgElement).toHaveAttribute('src');
-    expect(albumImgElement).toHaveAttribute('alt', album.id);
-
-    await waitFor(() => expect(albumImgElement).toHaveAttribute('src'));
-
-    expect(albumImgElement).toHaveAttribute('alt', album.id);
-    expect(apiMock.assetApi.getAssetThumbnail).not.toHaveBeenCalled();
-
-    expect(albumNameElement).toHaveTextContent(album.albumName);
-    expect(albumDetailsElement).toHaveTextContent(new RegExp(detailsText));
-  });
-
-  it('shows album data and loads the thumbnail image when available', async () => {
-    const thumbnailFile = new File([new Blob()], 'fileThumbnail');
-    const thumbnailUrl = 'blob:thumbnailUrlOne';
-    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-    //  @ts-ignore
-    // TODO: there needs to be a more robust mock of the @api to avoid mockResolvedValueOnce ts error
-    //       this is a workaround to make ts checks not fail but the test will pass as expected
-    apiMock.assetApi.getAssetThumbnail.mockResolvedValue({
-      data: thumbnailFile,
-      config: {},
-      headers: {},
-      status: 200,
-      statusText: '',
-    });
-    createObjectURLMock.mockReturnValueOnce(thumbnailUrl);
-
-    const album = albumFactory.build({
-      albumThumbnailAssetId: 'thumbnailIdOne',
-      shared: false,
-      albumName: 'some album name',
-    });
-    sut = render(AlbumCard, { album });
-
-    const albumImgElement = sut.getByTestId('album-image');
-    const albumNameElement = sut.getByTestId('album-name');
-    const albumDetailsElement = sut.getByTestId('album-details');
-    expect(albumImgElement).toHaveAttribute('alt', album.id);
-
-    await waitFor(() => expect(albumImgElement).toHaveAttribute('src', thumbnailUrl));
-
-    expect(albumImgElement).toHaveAttribute('alt', album.id);
-    expect(apiMock.assetApi.getAssetThumbnail).toHaveBeenCalledTimes(1);
-    expect(apiMock.assetApi.getAssetThumbnail).toHaveBeenCalledWith(
-      {
-        id: 'thumbnailIdOne',
-        format: ThumbnailFormat.Jpeg,
-      },
-      { responseType: 'blob' },
-    );
-    expect(createObjectURLMock).toHaveBeenCalledWith(thumbnailFile);
-
-    expect(albumNameElement).toHaveTextContent('some album name');
-    expect(albumDetailsElement).toHaveTextContent('0 items');
-  });
+  // it.each([
```

Comment:
why are these commented out? should we add a TODO to re-enable or should we delete them?

Reply:
Yeah, we can add a TODO. The tests rely on mocking the `src/api`, which is now gone.I have no experience with these style of frontend tests, so I have commented them out for now since I don't know how to fix them and can't be bothered at the moment.

Reply:
Hmm, the old mocks were mocking `getAssetThumbnail` and that API endpoint doesn't seem to be called anywhere in the codebase now. I'm a bit surprised by that. I haven't really checked to see how it was being called before, but it makes it a little tricky to give a recommendation on this. I can say that Vitest needs a function to be on an object in order to be able to mock it. We don't want to put all the API methods in a single object as that will prevent tree shaking. So we probably want something more like an object for each tested API method or an object for each tested Svelte file.

Reply:
Right. I brought this up originally when we did it that way in the CLI. We might just need to mock the whole `@immich/sdk` import or something like that.

Reply:
This specific album card component uses the new downlladRequest method via a xhr request to download the image.